app/controllers/discourse_subscriptions/hooks_controller.rb

```diff
@@ -17,66 +19,52 @@ def create
-        render_json_error e.message
-        return
+        return render_json_error e.message
       rescue ::Stripe::SignatureVerificationError => e
-        render_json_error e.message
-        return
+        return render_json_error e.message
       end
 
       case event[:type]
+      when 'customer.subscription.created'
       when 'customer.subscription.updated'
         customer = Customer.find_by(
           customer_id: event[:data][:object][:customer],
-          product_id: event[:data][:object][:plan][:product]
+          product_id: event[:data][:object][:plan][:product],
         )
 
-        if customer && subscription_completion?(event)
-          user = ::User.find(customer.user_id)
-          group = plan_group(event[:data][:object][:plan])
-          group.add(user) if group
-        end
+        return render_json_error 'customer not found' if !customer
+        return head 200 if event[:data][:object][:status] != 'complete'
 
-      when 'customer.subscription.deleted'
-        delete_subscription(event)
-      end
-
-      head 200
-    end
-
-    private
-
-    def subscription_completion?(event)
-      subscription_complete?(event) && previously_incomplete?(event)
-    end
-
-    def subscription_complete?(event)
-      event && event[:data] && event[:data][:object] && event[:data][:object][:status] && event[:data][:object][:status] == 'complete'
-    end
+        user = ::User.find_by(id: customer.user_id)
+        return render_json_error 'user not found' if !user
 
-    def previously_incomplete?(event)
-      event && event[:data] && event[:data][:previous_attributes] && event[:data][:previous_attributes][:status] && event[:data][:previous_attributes][:status] == 'incomplete'
-    end
+        if group = plan_group(event[:data][:object][:plan])
+          group.add(user)
+        end
+      when 'customer.subscription.deleted'
+        customer = Customer.find_by(
+          customer_id: event[:data][:object][:customer],
+          product_id: event[:data][:object][:plan][:product],
+        )
 
-    def delete_subscription(event)
-      customer = Customer.find_by(
-        customer_id: event[:data][:object][:customer],
-        product_id: event[:data][:object][:plan][:product]
-      )
+        return render_json_error 'customer not found' if !customer
 
-      if customer
-        sub_model = Subscription.find_by(
+        Subscription.find_by(
           customer_id: customer.id,
-          external_id: [:id]
-        )
-
-        sub_model.delete if sub_model
+          external_id: event[:data][:object][:id]
+        )&.destroy
 
         user = ::User.find(customer.user_id)
-        customer.delete
-        group = plan_group(event[:data][:object][:plan])
-        group.remove(user) if group
+        return render_json_error 'user not found' if !user
+
+        if group = plan_group(event[:data][:object][:plan])
+          group.remove(user)
+        end
+
+        customer.destroy
```

Comment:
```suggestion
        customer.destroy!
```

Rule of thumb: unless using the return boolean value (or noted another reason in a comment) use the bang version 😃